torchaudio/datasets/librispeech_finetune.py

```diff
@@ -0,0 +1,100 @@
+import os
+from pathlib import Path
+from typing import Tuple, Union
+
+import torchaudio
+from torch import Tensor
+from torch.hub import download_url_to_file
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import (
+    extract_archive,
+)
+
+
+FOLDER_IN_ARCHIVE = "librispeech_finetuning"
+URL = "https://dl.fbaipublicfiles.com/librilight/data/librispeech_finetuning.tgz"
+CHECKSUM = "5d1efdc777b548194d7e09ba89126e2188026df9fd57aa57eb14408d2b2342af"
+
+
+def _get_files(path, split, _ext_audio):
+    if split == "10min":
+        files = sorted(str(p) for p in Path(path).glob("1h/0/*/*/*/*" + _ext_audio))
+    elif split in ["1h", "10h"]:
+        files = [str(p) for p in Path(path).glob("1h/*/*/*/*/*" + _ext_audio)]
+        if split == "10h":
+            files += [str(p) for p in Path(path).glob("9h/*/*/*/*" + _ext_audio)]
+        files = sorted(files)
+    else:
+        raise ValueError(f"Unsupported split value. Found {split}.")
+    return files
+
+
+def _load_item(file_path: str, ext_audio: str, ext_txt: str) -> Tuple[Tensor, int, str, int, int, int]:
+    fileid = os.path.basename(file_path)
+    path = os.path.dirname(file_path)
+    speaker_id, chapter_id, utterance_id = fileid.replace(ext_audio, "").split("-")
+
+    file_text = speaker_id + "-" + chapter_id + ext_txt
+    file_text = os.path.join(path, file_text)
+
+    fileid_audio = speaker_id + "-" + chapter_id + "-" + utterance_id
+    file_audio = fileid_audio + ext_audio
+    file_audio = os.path.join(path, file_audio)
+
+    # Load audio
+    waveform, sample_rate = torchaudio.load(file_audio)
+
+    # Load text
+    with open(file_text) as ft:
+        for line in ft:
+            fileid_text, transcript = line.strip().split(" ", 1)
+            if fileid_audio == fileid_text:
+                break
+        else:
+            # Translation not found
+            raise FileNotFoundError("Translation not found for " + fileid_audio)
+
+    return (
+        waveform,
+        sample_rate,
+        transcript,
+        int(speaker_id),
+        int(chapter_id),
+        int(utterance_id),
+    )
+
+
+class LibriSpeechFineTune(Dataset):
```

Comment:
Although I understand that this dataset archive was created to facilitate fine-tuning, I am reluctant to call it `FineTune`, because the dataset itself can be used for other purposes.

I do not have an alternative suggestion, but the would it be possible to call something else?

Reply:
ESPnet uses [`LIBRILIGHT_LIMITED `](https://github.com/espnet/espnet/blob/master/egs2/librilight_limited/asr1/) as the dataset name (for the reference).